pandas/tests/indexing/interval/test_interval_new.py

```diff
@@ -143,9 +144,27 @@ def test_loc_with_slices(self):
         tm.assert_series_equal(expected, s[:2.5])
         tm.assert_series_equal(expected, s[0.1:2.5])
 
-        # slice of scalar with step != 1
-        with pytest.raises(ValueError):
-            s[0:4:2]
+    def test_slice_step_ne1(self):
+        # GH#31658 slice of scalar with step != 1
+        s = self.s
+        expected = s.iloc[0:4:2]
+
+        result = s[0:4:2]
+        tm.assert_series_equal(result, expected)
+
+        result2 = s[0:4][::2]
+        tm.assert_series_equal(result2, expected)
+
+    def test_slice_interval_step(self):
+        # GH#31658 allows for integer step!=1, not Interval step
+        s = self.s
+        msg = (
+            "cannot do slice indexing on "
+            "<class 'pandas.core.indexes.interval.IntervalIndex'> with "
+            "these indexers .* of <class 'pandas._libs.interval.Interval'>"
+        )
+        with pytest.raises(TypeError, match=msg):
+            s[0 : 4 : Interval(0, 1)]
```

Comment:
I would test it where start and stop are also intervals. 
And can you move it to the `test_loc_with_slices` a bit above that is testing slicing with Intervals ?

Reply:
> start and stop are also intervals

out of scope

>  And can you move it to the test_loc_with_slices a bit above that is testing slicing with Intervals ?

sure

Reply:
> > start and stop are also intervals
> 
> out of scope

I think that is not out of scope, as that hits this path, and right now actually ensured there is a proper error message.